internal/validator/validator_test.go

```diff
@@ -0,0 +1,256 @@
+				}
+			}
+		})
+	}
+}
+
+// TestCheckMessageScope tests the checkMessageScope function.
+func TestCheckMessageScope(t *testing.T) {
+	testCases := []struct {
+		name          string
+		scope         string
+		expectedError string
+	}{
+		{
+			name:          "Valid Scope - Lowercase",
+			scope:         "user",
+			expectedError: "",
+		},
+		{
+			name:          "Valid Scope - Empty",
+			scope:         "",
+			expectedError: "",
+		},
+		{
+			name:  "Invalid Scope - Uppercase",
+			scope: "User",
+			expectedError: `invalid commit message scope casing, "User" ` +
+				`should be "user"`},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := checkMessageScope(&tc.scope)
+			if tc.expectedError == "" {
+				if err != nil {
+					t.Errorf("expected no error, got %v", err)
+				}
+			} else {
+				if err == nil || err.Error() != tc.expectedError {
+					t.Errorf("expected error %q, got %q", tc.expectedError, err)
+				}
+			}
+		})
+	}
+}
+
+// TestCheckMessageSubject tests the checkMessageSubject function.
+func TestCheckMessageSubject(t *testing.T) {
+	testCases := []struct {
+		name          string
+		subject       string
+		expectedError string
+	}{
+		{
+			name:          "Valid Subject - Lowercase",
+			subject:       "add a new feature",
+			expectedError: "",
+		},
+		{
+			name:    "Invalid Subject - Uppercase",
+			subject: "Add a new feature",
+			expectedError: "commit message subject should be lowercased & " +
+				"not end with a period(.)",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := checkMessageSubject(&tc.subject)
+			if tc.expectedError == "" {
+				if err != nil {
+					t.Errorf("expected no error, got %v", err)
+				}
+			} else {
+				if err == nil || err.Error() != tc.expectedError {
+					t.Errorf("expected error %q, got %q", tc.expectedError, err)
+				}
+			}
+		})
+	}
+}
+
+// TestCheckMessageLength tests the checkMessageLength function.
+func TestCheckMessageLength(t *testing.T) {
+	testCases := []struct {
+		name          string
+		message       string
+		expectedError string
+	}{
+		{
+			name:          "Valid Length",
+			message:       "feat: add a new feature",
+			expectedError: "",
+		},
+		{
+			name: "Invalid Length",
+			message: "Add a new feature that makes the application run " +
+				"faster and more efficiently",
+			expectedError: fmt.Sprintf(
+				"commit message exceeds 50 characters, current length: %d",
+				len("Add a new feature that makes the application run "+
+					"faster and more efficiently"),
+			),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := checkMessageLength(&tc.message)
+			if tc.expectedError == "" {
+				if err != nil {
+					t.Errorf("expected no error, got %v", err)
+				}
+			} else {
+				if err == nil || err.Error() != tc.expectedError {
+					t.Errorf("expected error %q, got %q", tc.expectedError, err)
+				}
```

Comment:
The conditional check for the error message can be much simplified if the `expectedError` is of the `error` interface type, so refactor it after a thorough evaluation of the aforementioned comments.